### ISSUE TITLE
Error occured

### ISSUE THREAD

**Issue by NickSackett** [Arckenver/Nations#57]
getting the error occurred while exexcuting command: no value present.
Nations-1.1.jar
spongeforge-1.8.9-1890-4.2.0-BETA-1653.jar

**Comment by liach** [comment]
Stacktrace?

**Comment by NickSackett** [comment]
what do you mean exactly?

**Comment by liach** [comment]
Go to the backend of the server, reproduce the error, put the backend log onto sites like https://gist.github.com, and provide a link to that gist.

**Comment by NickSackett** [comment]
https://gist.github.com/NickSackett/c7795005bcf76219bdb1456a2d1adadd

**Comment by liach** [comment]
Here I will randomly pick some revelant lines:

```
[22:42:02] [Server thread/ERROR] [Sponge/]: Error occurred while executing command 'na perm MineHavenCity citizen build 0' for source EntityPlayerMP['Nick_Sackett'/374406, l='world', x=-735.61, y=72.00, z=175.45]: No value present
java.util.NoSuchElementException: No value present
at java.util.Optional.get(Optional.java:135) ~[?:1.8.0_101]
at com.arckenver.nations.ConfigHandler.getNationRank(ConfigHandler.java:190) ~[ConfigHandler.class:1.1]
at com.arckenver.nations.Utils.formatNationDescription(Utils.java:89) ~[Utils.class:1.1]
at com.arckenver.nations.cmdexecutor.nationadmin.NationadminPermExecutor.execute(NationadminPermExecutor.java:38) ~[NationadminPermExecutor.class:1.1]
at org.spongepowered.api.command.args.ChildCommandElementExecutor.execute(ChildCommandElementExecutor.java:177) ~[ChildCommandElementExecutor.class:1.8.9-1890-4.2.0-BETA-1653]
at org.spongepowered.api.command.spec.CommandSpec.process(CommandSpec.java:331) ~[CommandSpec.class:1.8.9-1890-4.2.0-BETA-1653]
at org.spongepowered.api.command.dispatcher.SimpleDispatcher.process(SimpleDispatcher.java:331) ~[SimpleDispatcher.class:1.8.9-1890-4.2.0-BETA-1653]
at org.spongepowered.common.command.SpongeCommandManager.process(SpongeCommandManager.java:279) [SpongeCommandManager.class:1.8.9-1890-4.2.0-BETA-1653]
at net.minecraft.command.ServerCommandManager.func_71556_a(SourceFile:80) [bd.class:?]
at net.minecraft.network.NetHandlerPlayServer.func_147361_d(NetHandlerPlayServer.java:815) [lm.class:?]
at net.minecraft.network.NetHandlerPlayServer.func_147354_a(NetHandlerPlayServer.java:794) [lm.class:?]
at net.minecraft.network.play.client.C01PacketChatMessage.func_148833_a(SourceFile:37) [ie.class:?]
at net.minecraft.network.play.client.C01PacketChatMessage.func_148833_a(SourceFile:9) [ie.class:?]
at org.spongepowered.common.network.PacketUtil.onProcessPacket(PacketUtil.java:121) [PacketUtil.class:1.8.9-1890-4.2.0-BETA-1653]
at net.minecraft.network.PacketThreadUtil$1.redirect$onProcessPacket$0(SourceFile:39) [fh$1.class:?]
at net.minecraft.network.PacketThreadUtil$1.run(SourceFile:13) [fh$1.class:?]
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_101]
at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_101]
at net.minecraft.util.Util.func_181617_a(SourceFile:44) [g.class:?]
at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:660) [MinecraftServer.class:?]
at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:344) [ko.class:?]
at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:605) [MinecraftServer.class:?]
at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:481) [MinecraftServer.class:?]
at java.lang.Thread.run(Thread.java:745) [?:1.8.0_101]
```

Looks like a bad handling of `Optional`s.

**Comment by NickSackett** [comment]
what does that mean?

**Comment by liach** [comment]
Generally bad coding style.

**Comment by NickSackett** [comment]
what am i suppose to do?

**Comment by liach** [comment]
Wait for reply or fix from @Arckenver.

**Comment by Arckenver** [comment]
Here I guess you don't have enough `nationRanks` in your config and the plugin can't find the nation rank of that nation. Can you paste the content of your config.conf ?

But anyway, @TheoKah made a [PR](https://github.com/Arckenver/Nations/pull/50) that fixed that error recently. It'll be available in the next release of Nations.

**Comment by NickSackett** [comment]
```
nationRanks=[
    {
        nationTitle=Land
        numCitizens=1
        presidentTitle=Leader
    },
    {
        nationTitle=Federation
        numCitizens=3
        presidentTitle=Count
    },
    {
        nationTitle=Dominion
        numCitizens=6
        presidentTitle=Duke
    },
    {
        nationTitle=Kingdom
        numCitizens=10
        presidentTitle=King
    },
    {
        nationTitle=Empire
        numCitizens=15
        presidentTitle=Emperor
    }
]
nations {
    canEditTaxes=true
    defaultTaxes=50
    flags {
        explosions=false
        fire=false
        mobs=false
        pvp=false
    }
    maxTaxes=100
    perms {
        citizen {
            build=false
            interact=true
        }
        outsider {
            build=false
            interact=false
        }
    }
}
others {
    blocksPerCitizen=10000
    enableNationRanks=true
    maxExtra=5000
    maxNationNameLength=13
    maxNationSpawns=3
    maxZoneNameLength=13
    minNationDistance=500
    minNationNameLength=3
    minZoneNameLength=3
}
prices {
    blockClaimPrice=0.3
    extraPrice=0.5
    nationCreationPrice=2500
    outpostCreationPrice=1000
    unclaimRefundPercentage=0
    upkeepPerCitizen=100
}
worlds {
    DIM-1 {
        enabled=true
        flags {
            explosions=true
            fire=true
            mobs=true
            pvp=true
        }
        perms {
            build=true
            interact=true
        }
    }
    DIM1 {
        enabled=true
        flags {
            explosions=true
            fire=true
            mobs=true
            pvp=true
        }
        perms {
            build=true
            interact=true
        }
    }
    world {
        enabled=true
        flags {
            explosions=true
            fire=true
            mobs=true
            pvp=true
        }
        perms {
            build=true
            interact=true
        }
    }
}
zones {
    perms {
        citizen {
            build=false
            interact=true
        }
        coowner {
            build=true
            interact=true
        }
        outsider {
            build=false
            interact=false
        }
    }
}
```

**Comment by Arckenver** [comment]
This error has been fixed in [v2.2](https://github.com/Arckenver/Nations/releases/tag/v2.2) of Nations (API 5). A fix for v1.x (API 4) will take more time.
